tests/ftqc/test_parametric_mid_measure.py

```diff
@@ -624,100 +624,42 @@ def test_diagonalize_mcm_transform_preserves_reset(self, reset):
         assert isinstance(new_tape.operations[-1], MidMeasureMP)
         assert new_tape.operations[-1].reset == reset
 
-    def test_diagonalize_mcm_in_cond(self):
+    def test_diagonalize_mcm_applied_in_cond(self):
         """Test that the diagonalize_mcm transform works as expected on a tape
-        containing a conditional with a ParametricMidMeasureMP"""
+        conditionally applying two ParametricMidMeasureMPs as the true and false
+        condition respectively"""
 
         with qml.queuing.AnnotatedQueue() as q:
             qml.RX(1.2, 0)
             m = qml.measure(0)
-            qml.cond(m == 1, partial(ParametricMidMeasureMP, angle=1.2))(wires=2, plane="XY")
+            qml.cond(
+                m == 1,
+                partial(ParametricMidMeasureMP, angle=1.2),
+                partial(ParametricMidMeasureMP, angle=-1.2),
+            )(wires=2, plane="XY")
 
         original_tape = qml.tape.QuantumScript.from_queue(q)
-        base_diagonalizing_gates = ParametricMidMeasureMP(
-            Wires([2]), angle=1.2, plane="XY"
-        ).diagonalizing_gates()
+        base_diagonalizing_gates = (
+            ParametricMidMeasureMP(Wires([2]), angle=1.2, plane="XY").diagonalizing_gates()
+            + ParametricMidMeasureMP(Wires([2]), angle=-1.2, plane="XY").diagonalizing_gates()
+        )
 
         (new_tape,), _ = diagonalize_mcms(qml.tape.QuantumScript.from_queue(q))
-        assert len(new_tape.operations) == 5
-        diagonalizing_gates = new_tape.operations[2:4]
-        measurement = new_tape.operations[4]
+        assert len(new_tape.operations) == 7
+        diagonalizing_gates = new_tape.operations[2:6]
+        measurement = new_tape.operations[6]
 
         # conditional diagonalizing gate
         for gate, expected_base in zip(diagonalizing_gates, base_diagonalizing_gates):
             assert isinstance(gate, qml.ops.Conditional)
             assert gate.base == expected_base
             assert gate.wires == original_tape.operations[2].wires
 
-        # conditional diagonalized ParametricMidMeasureMP
-        assert isinstance(measurement, qml.ops.Conditional)
-        assert measurement.wires == original_tape.operations[2].wires
-        assert isinstance(measurement.base, MidMeasureMP)
-        assert not isinstance(measurement.base, ParametricMidMeasureMP)
+        # diagonalized ParametricMidMeasureMP
+        assert isinstance(measurement, MidMeasureMP)
+        assert not isinstance(measurement, ParametricMidMeasureMP)
 
-        # cond(diagonalizing gate) and cond(mcm) rely on same measurement in the same way
-        assert (
-            diagonalizing_gates[0].meas_val.measurements[0] == measurement.meas_val.measurements[0]
-        )
-        assert diagonalizing_gates[0].meas_val.processing_fn == measurement.meas_val.processing_fn
-
-    def test_diagonalize_mcm_cond_two_outcomes(self):
```

Comment:
There's currently only support for conditional having two measurement outcomes, so this test and the previous test became the same test